File: network/https-server/https-server.py
#! /usr/bin/python3

import ssl
from http.server import HTTPServer, SimpleHTTPRequestHandler

# bind to all interfaces, use port 4433
httpd = HTTPServer(('0.0.0.0', 4433), SimpleHTTPRequestHandler)

context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
context.load_cert_chain("cert.pem", "key.pem")

httpd.socket = context.wrap_socket(httpd.socket, server_side=True)
print("Starting HTTPS server on port 4433...")
httpd.serve_forever()

File: network/https-server/https-server.rb
#! /usr/bin/env ruby

require "openssl"
require "webrick"

cert = File.join(__dir__, "cert.pem")
key = File.join(__dir__, "key.pem")

server = WEBrick::HTTPServer.new(
  :DocumentRoot => ".",
  :Port => 4433,
  :SSLEnable => true,
  :SSLCertificate => OpenSSL::X509::Certificate.new(File.read(cert)),
  :SSLPrivateKey => OpenSSL::PKey::RSA.new(File.read(key))
)

# cleanly shutdown the server after pressing Ctrl+C
trap "INT" do
  server.shutdown
end

puts "Starting HTTPS server on port 4433..."
server.start
